django-rgd/rgd/rest/viewsets.py

```diff
@@ -18,6 +19,24 @@ class CollectionViewSet(ModelViewSet):
     queryset = models.Collection.objects.all()
     filterset_class = CollectionFilter
 
+    @swagger_auto_schema(
+        method='GET',
+        operation_summary='Get the file at the given index in this collection. Associated files are orderd by primary key and the provided index is the index in that ordered set. If the files in the collection change, this will not be reproducible.',
+    )
+    @action(
+        detail=True,
+        serializer_class=serializers.ChecksumFileSerializer,
+        url_path=r'item/(?P<index>\d+)',
+    )
+    def item(self, request, pk, index):
+        collection = models.Collection.objects.get(pk=pk)
+        files = collection.checksumfiles.order_by('pk')
+        try:
+            instance = files[int(index)]
+        except (IndexError, ValueError):
+            raise APIException(f'index {index} not valid or out of range.')
```

Comment:
When I test this locally I'm seeing this exception come back as a 500 error - is there a way to configure `APIException` to raise a 400 error instead? Or perhaps
```suggestion
            raise Http404(f'index {index} not valid or out of range.')
```
Would be better? An invalid index is strictly a user error and not a server error so I think 400 is more appropriate.

Reply:
Hm, good catch on the error code. I added a new `APIException` subclass for this in d54ea26 that will 404

Reply:
Scratch that, `from rest_framework.exceptions import ValidationError` seems like the right exception